handler/file/retrieve.go

```diff
@@ -114,13 +114,81 @@ func (r *filecoinReader) Read(p []byte) (int, error) {
 		readLen = remainingBytes
 	}
 
+	buf := bytes.NewBuffer(p)
+	buf.Reset()
+
+	n, err := r.writeToN(buf, readLen)
+	return int(n), err
+}
+
+func (r *filecoinReader) WriteTo(w io.Writer) (int64, error) {
+	if r.offset >= r.size {
+		return 0, io.EOF
+	}
+	// Read all remaining bytes and write them to w.
+	return r.writeToN(w, r.size-r.offset)
+}
+
+func (r *filecoinReader) writeToN(w io.Writer, readLen int64) (int64, error) {
+	// Check if offset is in current range.
+	var read int64
+	if r.rangeReader != nil {
+		if r.rangeReader.inRange(r.offset) {
+			// If seek happened since last partial range read. A backwards seek
+			// or a seek beyond the remaining data would not be in range.
```

Comment:
```suggestion
			// If seek happened since last partial range read. A backward seek
			// or a seek beyond the remaining data would not be in range.
			//
			// TODO: If the forward seek is sufficiently large, it may be more
			// efficient to re-retrieve the data than to discard it from this
			// rangeReader.
```